src/sessions.ts

```diff
@@ -748,6 +863,11 @@ export class ServerSessionPool {
    */
   release(session: ServerSession): void {
     const sessionTimeoutMinutes = this.topology.logicalSessionTimeoutMinutes;
+
+    if (this.topology.loadBalanced) {
+      this.sessions.unshift(session);
```

Comment:
Is it possible to have a double unshift here? we're falling through to the logic beneath which could call unshift again

Reply:
Can we be in loadbalanced mode with a non-zero logicalSessionTimeoutMinutes (this should always be non-zero in LB) and the current session has not timed out and isn't dirty? If so then we're duplicating the insert if I'm reading the booleans correctly.

Reply:
Ah ok yea I can see the the logicalSessionTimeoutMinutes isn't defined. I agree we should assert something here I think then

Reply:
I've added the assertion that `sessionTimeoutMinutes` also does not exist.